Gradle: prefer local deps as JARs when building runnable JARs

aloubyansky opened the pull request:
When building runnable JARs using Gradle, dependencies from the same project/workspace should be consumed as JARs, not as directories of classes

Fixes #7365

gastaldi reviewed (approved):
Looks good, but can we add a test too?

aloubyansky committed:
When building runnable JARs using Gradle, dependencies from the same project/workspace should be consumed as JARs, not as directories of classes.

Update devtools/gradle/src/functionalTest/java/io/quarkus/gradle/BasicMultiModuleTest.java

Co-Authored-By: George Gastaldi <gegastaldi@gmail.com>

aloubyansky commented:
Added a test.

gastaldi commented on devtools/gradle/src/functionalTest/java/io/quarkus/gradle/BasicMultiModuleTest.java:
```diff
@@ -0,0 +1,48 @@
+                .build();
+        
+        final Path commonLibs = projectDir.toPath().resolve("common").resolve("build").resolve("libs");
+        assertThat(Files.exists(commonLibs)).isEqualTo(true);
```
```suggestion
        assertThat(commonLibs).exists();
```

gastaldi commented on devtools/gradle/src/functionalTest/java/io/quarkus/gradle/BasicMultiModuleTest.java:
```diff
@@ -0,0 +1,48 @@
+        
+        final Path commonLibs = projectDir.toPath().resolve("common").resolve("build").resolve("libs");
+        assertThat(commonLibs).exists();
+        assertThat(Files.exists(commonLibs.resolve("common.jar"))).isEqualTo(true);
```
Same as above

gastaldi commented:
I rebased and force-pushed your branch with the test changes. Looks great!
